packages/console/src/containers/AppContent/components/Topbar/TenantSelector/components/CreateTenantModal/CreateForm.tsx

```diff
@@ -0,0 +1,107 @@
+import type { AdminConsoleKey } from '@logto/phrases';
+import { type TenantInfo, type CreateTenant, TenantTag } from '@logto/schemas';
+import { Controller, FormProvider, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import Close from '@/assets/images/close.svg';
+import HeaderIcon from '@/assets/images/create-tenant-header.svg';
+import { useCloudApi } from '@/cloud/hooks/use-cloud-api';
+import Button from '@/components/Button';
+import Card from '@/components/Card';
+import CardTitle from '@/components/CardTitle';
+import FormField from '@/components/FormField';
+import IconButton from '@/components/IconButton';
+import RadioGroup, { Radio } from '@/components/RadioGroup';
+import TextInput from '@/components/TextInput';
+
+import * as styles from './index.module.scss';
+
+const tagOptions: Array<{ title: AdminConsoleKey; value: TenantTag }> = [
+  {
+    title: 'tenants.create_modal.environment_tag_development',
+    value: TenantTag.Development,
+  },
+  {
+    title: 'tenants.create_modal.environment_tag_staging',
+    value: TenantTag.Staging,
+  },
+  {
+    title: 'tenants.create_modal.environment_tag_production',
+    value: TenantTag.Production,
+  },
+];
+
+type Props = {
+  onClose: (tenant?: TenantInfo) => void;
+};
+
+type CreateTenantPayload = Required<Pick<CreateTenant, 'name' | 'tag'>>;
+
+export function CreateForm({ onClose }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const methods = useForm<CreateTenantPayload>({ defaultValues: { tag: TenantTag.Development } });
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+    register,
+  } = methods;
+
+  const cloudApi = useCloudApi();
+
+  const onSubmit = handleSubmit(async (data) => {
+    const { name, tag } = data;
+    const newTenant = await cloudApi
+      .post('/api/tenants', { json: { name, tag } })
+      .json<TenantInfo>();
+    onClose(newTenant);
+  });
```

Comment:
Might still need to use `try catch` here due to the react hook form change we discussed last time?

Reply:
updated